Content.Server/Chat/Systems/ChatSystem.Radio.cs

```diff
@@ -45,6 +71,25 @@ private void ShutdownRadio()
         // TODO: Turn common into a true frequency and support multiple aliases.
         var isRadioMessage = false;
         RadioChannelPrototype? channel = null;
+
+        // Replaces :h with whatevs the headset's "default" is
+        var locChannelMessage = message.StartsWith(":h") || message.StartsWith(".h");
+        if (locChannelMessage && message.Length >= 2)
+        {
+            if (_inventory.TryGetSlotEntity(source, "ears", out var headsetUid) && HasComp<HeadsetComponent>(headsetUid))
+            {
```

Comment:
This method should only be concerned with trimming the message and not grabbing headset like this.

Reply:
This part I need to think about... the problem is that we need to get the headset component in order to grab its yml, with defaultChannel being YML ideally. But it can only do this changing around of message keycodes if it knows said defaultChannel

Would it be better to add the defaultChannel under the "h" keycode, and have the code processed the same way? I'd still need to grab the headset's comp so I can see the defaultChannel value

Reply:
I think my latest commit fixes this. We're still having to grab the headset, but we already did that for "hasHeadset". So I thought it would be okay if we moved "hasHeadset" up and use "TryComp" instead of "HasComp" so we can grab the Comp later it would be okay to do

Reply:
I've cleaned up the code considerably as I've learned a lot the last few days. The TryComp I used is confirmed to not do anything more than HasComp, but stores the output so I think it's good now! The original code used HasComp already